common/djangoapps/embargo/models.py

```diff
@@ -163,13 +163,29 @@ def is_restricted_course(cls, course_id):
         return unicode(course_id) in cls._get_restricted_courses_from_cache()
 
     @classmethod
+    def is_disabled_access_check(cls, course_id):
+        """
+        Check if the course is in restricted list has disabled_access_check
+
+        Args:
+            course_id (str): course_id to look for
+
+        Returns:
+            Boolean
+            disabled_access_check attribute of restricted course
+        """
+        return cls._get_restricted_courses_from_cache().get(unicode(course_id), False)
+
+    @classmethod
     def _get_restricted_courses_from_cache(cls):
         """
-        Cache all restricted courses and returns the list of course_keys that are restricted
+        Cache all restricted courses and returns the dict of course_keys and disable_access_check that are restricted
         """
         restricted_courses = cache.get(cls.COURSE_LIST_CACHE_KEY)
         if restricted_courses is None:
-            restricted_courses = list(RestrictedCourse.objects.values_list('course_key', flat=True))
+            restricted_courses = dict(
+                RestrictedCourse.objects.values_list('course_key', 'disable_access_check')
+            )
```

Comment:
I'm a little concerned about the extensibility of this approach.  What happens if we need to add additional information for the restricted course in the future?

Maybe something like this would be more extensible?
```
{
    course.course_key: {
        'disable_access_check': course.disable_access_check
    }
    for course in RestrictedCourse.objects.all()
}
```